database_wrapper.py

```diff
@@ -630,23 +634,41 @@ def update_entity(self, entity_type, data_dict: dict, filter_fields: list) -> bo
             query = query.filter(getattr(entity_type, field) == formatted_data[field])
         entity = query.first()
 
+        # Logging...
         if entity:
-            print("{}Updating {} in {}...{}".format(
-                  YELLOW_COLOR_CODE, entity, entity_attributes['__tablename__'], RESET_COLOR_CODE))
+            print(
+                "{}Updating {} in {}...{}".format(
+                    YELLOW_COLOR_CODE,
+                    entity,
+                    entity_attributes["__tablename__"],
+                    RESET_COLOR_CODE,
+                )
+            )
         else:
             entity = entity_type()
-            print("{}Matching Entity not found - Inserting {} in {}...{}".format(
-                  YELLOW_COLOR_CODE, entity, entity_attributes['__tablename__'], RESET_COLOR_CODE))
+            print(
+                "{}Matching Entity not found - Inserting {} in {}...{}".format(
+                    YELLOW_COLOR_CODE,
+                    entity,
+                    entity_attributes["__tablename__"],
+                    RESET_COLOR_CODE,
+                )
+            )
 
-        # Grab the entity class fields by cleaning the attributes dictionary - 
-        # discard anything with underscores in the front or back
+        # Grab the entity class fields by cleaning the attributes dictionary
         # Note: Make sure you don't label any important data fields with underscores in the front or back!
-        entity_fields = dict(filter(lambda i: not (i[0][0] == '_' or i[0][-1] == '_'), 
-                                    entity_attributes.items()))
+        entity_fields = list(
+            dict(
+                filter(
+                    lambda i: not (i[0][0] == "_" or i[0][-1] == "_"),
+                    entity_attributes.items(),
+                )
+            ).keys()
+        )[1:]
 
         # Ignore the first field, since it's assumed to be a primary key
         # Populate the entity with values from formatted_data
-        for entity_field in list(entity_fields.keys())[1:]:
+        for entity_field in entity_fields:
```

Comment:
This no longer ignores the first field - should we update the comment?

Reply:
Right - I'll make a quick edit to this comments. Thanks for the catch!